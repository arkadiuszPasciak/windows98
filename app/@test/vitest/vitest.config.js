import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
