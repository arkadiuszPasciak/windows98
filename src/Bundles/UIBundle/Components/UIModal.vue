<template>
  <div
    class="UIModal"
    :class="`variant-${variant}`"
    :style="`width: 800px; height: 800px;`"
  >
    <div class="header">
      <h3 class="title">{{ title }}</h3>
      <UIButton class="button-close">
        x
        <!-- <template #icon-left>
          <img
            class="button-icon"
            :width="23"
            :height="23"
            src="src of image"
            alt="close window"
          />
        </template> -->
      </UIButton>
    </div>
    <div class="content">
      <slot />
    </div>
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'

  defineProps({
    variant: {
      type: String as PropType<'primary'>,
      default: 'primary',
    },
    title: {
      type: String,
      required: true,
    },
  })
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UIModal.styles.scss';
</style>
