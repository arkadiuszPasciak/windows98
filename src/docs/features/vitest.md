# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
