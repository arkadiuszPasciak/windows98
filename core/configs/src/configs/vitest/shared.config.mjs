import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
