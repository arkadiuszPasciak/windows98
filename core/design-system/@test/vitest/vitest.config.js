import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
