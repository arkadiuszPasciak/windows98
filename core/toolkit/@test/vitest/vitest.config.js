import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
