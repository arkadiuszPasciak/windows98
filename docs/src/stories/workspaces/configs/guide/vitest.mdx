import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

