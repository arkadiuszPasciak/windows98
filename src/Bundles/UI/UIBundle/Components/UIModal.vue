<template>
  <div
    v-if="modalState"
    ref="modalElement"
    class="UIModal"
    :class="classes"
    @mousemove="(event) => mouseMove(event)"
  >
    <div class="container" :style="{ width: `${width}`, height: `${height}` }">
      <div
        class="header"
        :style="`cursor: ${cursorType}`"
        @mousedown="(event) => mouseDownEvent(event)"
        @mouseup="mouseUpEvent"
      >
        <h3 class="title" data-test="modal-header-title">{{ title }}</h3>
        <UIButton
          class="button-close"
          size="small"
          data-test="ui-modal-button-close"
          @click="$emit('closeModal')"
        >
          <template #icon-left>
            <img
              class="button-close-icon"
              :width="8"
              :height="7"
              src="/icons/close-modal.svg"
              :alt="t('UIBundle.close-window')"
            />
          </template>
        </UIButton>
      </div>

      <UIModalNavigation v-if="$slots.options">
        <slot name="options" />
      </UIModalNavigation>

      <slot />
    </div>
  </div>
</template>

<script setup lang="ts">
  import { Ref, ref } from 'vue'
  import { Nullable } from 'vitest'
  import { useI18n } from 'vue-i18n'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'
  import UIModalNavigation from '@Bundles/UIBundle/Components/UIModalNavigation.vue'
  import { IUIModalCursorType } from '@Bundles/UIBundle/Supports/UIModal.supports'

  const props = defineProps({
    title: {
      type: String,
      required: true,
    },
    width: {
      type: String,
      default: `450px`,
    },
    height: {
      type: String,
      default: `150px`,
    },
    resizeWindow: {
      type: Boolean,
      default: false,
    },
    moveWindow: {
      type: Boolean,
      default: true,
    },
    modalState: {
      type: Boolean,
      default: false,
    },
  })

  const { t } = useI18n()

  const classes = [
    `${props.resizeWindow ? 'resize-window' : ''}`,
    `${props.moveWindow ? 'move-window' : ''}`,
  ] as Array<string>

  const modalElement = ref(null) as Ref<Nullable<HTMLElement>>
  const mouseState = ref(false) as Ref<boolean>
  const positionX = ref(0) as Ref<number>
  const positionY = ref(0) as Ref<number>
  const cursorType = ref('default') as Ref<IUIModalCursorType>

  const mouseDownEvent = (event: MouseEvent): void => {
    if (!modalElement.value || !props.moveWindow) {
      return
    }

    cursorType.value = 'move'
    mouseState.value = true
    positionX.value = modalElement.value.offsetLeft - event.clientX
    positionY.value = modalElement.value.offsetTop - event.clientY
  }

  const mouseUpEvent = (): void => {
    if (!props.moveWindow) {
      return
    }

    mouseState.value = false
    cursorType.value = 'default'
  }

  const mouseMove = (event: MouseEvent): void => {
    if (!mouseState.value || !modalElement.value || !props.moveWindow) {
      return
    }

    modalElement.value.style.left = event.clientX + positionX.value + 'px'
    modalElement.value.style.top = event.clientY + positionY.value + 'px'
  }

  defineEmits(['closeModal'])
</script>

<i18n src="@Bundles/UIBundle/Locales/UIBundle.locales.json"></i18n>

<style lang="scss" scoped src="@Bundles/UIBundle/Styles/UIModal.styles.scss" />
