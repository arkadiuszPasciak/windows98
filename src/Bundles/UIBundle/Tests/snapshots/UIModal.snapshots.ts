import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'

const defaultWrapper = mount(UIModal)

describe('[UIBundle]<Snapshots>(UIModal)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
