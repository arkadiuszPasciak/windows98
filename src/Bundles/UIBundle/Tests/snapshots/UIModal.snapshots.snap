// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UIModal) > renders correctly component 1`] = `
<div
  class="UIModal variant-primary move-window"
  data-v-99a75817=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 500px; height: 300px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-v-99a75817=""
      >
        Close Program
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    
    
  </div>
</div>
`;
